docs/design/contextual-tabs.md

```diff
@@ -0,0 +1,392 @@
+>
+> * Excel on Windows (Microsoft 365 only, not perpetual license): Version ???? (Build ?????.?????) Your Microsoft 365 subscription may need to be on the [Current Channel (Preview)](https://insider.office.com/en-us/join/windows) formerly called "Monthly Channel (Targeted)" or "Insider Slow".
+
+> [!NOTE]
+> Custom contextual tabs work only on platforms that support the following requirement sets. For more about requirement sets and how to work with them, see [Specify Office applications and API requirements](../develop/specify-office-hosts-and-api-requirements.md).
+>
+> - [SharedRuntime 1.1](../reference/requirement-sets/shared-runtime-requirement-sets.md)
+> - [RibbonApi 1.2](../reference/requirement-sets/ribbon-api-requirement-sets.md)
+
+## Behavior of custom contextual tabs
+
+The user experience for custom contextual tabs follows the pattern of built-in Office contextual tabs. The following are the basic principles for the placement custom contextual tabs:
+
+- When a custom contextual tab is visible, it appears on the right end of the ribbon.
+- If one or more built-in contextual tabs and one or more custom contextual tabs from add-ins are visible at the same time, the custom contextual tabs are always to the right of all of the built-in contextual tabs.
+- If your add-in has more than one contextual tab and there are contexts in which more than one is visible, they appear in the order (left-to-right) in which they are defined in your add-in. See [Define the groups and controls that appear on the tab](#define-the-groups-and-controls-that-appear-on-the-tab) for details about how you define them.
+- If more than one add-in has a contextual tab that is visible in a specific context, then they appear in the order (left-to-right) in which the add-ins were installed.
+
+## Major steps for including a contextual tab in an add-in
+
+The following are the major steps for including a custom contextual tab in an add-in:
+
+1. Configure the add-in to use a shared runtime.
+1. Define the tab and the groups and controls that appear on it.
+1. Register the contextual tab with Office.
+1. Specify the circumstances when the tab will be visible.
+
+## Configure the add-in to use a shared runtime
+
+Adding custom contextual tabs requires your add-in to use the shared runtime. For more information, [Configure an add-in to use a shared runtime](../excel/configure-your-add-in-to-use-a-shared-runtime.md).
+
+## Define the groups and controls that appear on the tab
+
+Unlike custom core tabs, which are defined with XML in the manifest, custom contextual tabs are defined at runtime with a JSON blob. With code that runs when the add-in starts, you parse the blob into a JavaScript object, and then pass the object to the [Office.ribbon.requestCreateControls](/javascript/api/office/office.ribbon?view=common-js&preserve-view=true#requestcreateconrtols-input-) method. 
```

Comment:
the call to Office.ribbon.requestCreateControls can be done at any time during or after initialization of the Web Add-in.  But it can only be successful 1 time per document.